### DISCUSSION TITLE
How do I set up socks5 while using upstream？

### DISCUSSION THREAD

**Discussion by shamrock33** [mitmproxy/mitmproxy#5320] [Q&A]
I know you can use upstream to forward HTTP traffic, for example
--mode upstream:http://localhost:11001/

Want to forward sock5 traffic should be how to write
I tried - mode upstream: socks5: / / localhost: 11001 / such as not

**Comment by mhils** [comment]
Upstream Proxy Mode and SOCKS proxying are mutually exclusive at the moment.

**Reply by shamrock33** [reply]
So upstream proxy mode I can only connect to HTTP proxy right？

**Reply by mhils** [reply]
Correct.

**Reply by davidwang1953** [reply]
How should I understand that “upstream proxy mode and socks proxy mode are mutually exclusive”, the current upstream mode only supports http protocol, can I extend it to support socks5 protocol, is there any obstacle to it? Looking forward to your reply
